python/nav/web/navlets/__init__.py

```diff
@@ -243,24 +244,40 @@ def dispatcher(request, navlet_id):
     The as_view method takes any attribute and adds it to the instance
     as long as it is defined on the Navlet class
     """
-    account = request.account
+    current_account = request.account
+
     try:
-        account_navlet = AccountNavlet.objects.get(account=account, pk=navlet_id)
+        account_navlet = AccountNavlet.objects.get(pk=navlet_id)
     except AccountNavlet.DoesNotExist as error:
         _logger.error(
-            '%s tried to fetch widget with id %s: %s', account, navlet_id, error
+            '%s tried to fetch widget with id %s: %s', current_account, navlet_id, error
         )
         return HttpResponse(status=404)
-    else:
-        cls = get_navlet_from_name(account_navlet.navlet)
-        if not cls:
-            cls = get_navlet_from_name(ERROR_WIDGET)
-        view = cls.as_view(
-            preferences=account_navlet.preferences,
-            navlet_id=navlet_id,
-            account_navlet=account_navlet,
+
+    dashboard = account_navlet.dashboard
+    owner = account_navlet.account
+    can_edit = dashboard.can_edit(current_account)
+    is_shared = dashboard.is_shared
+
+    if not can_edit and not is_shared:
```

Comment:
So shared but not editable widgets is not a thing?

Reply:
But this could of course have been an OR, `if can_edit or is_shared`

Reply:
Changed the if check to `if not(can_edit or is_shared):` because I find it clearer.